src/strace.c

```diff
@@ -1515,6 +1518,27 @@ startup_attach(void)
 	}
 }
 
+static bool
+is_uid_gid_pair(const char *username, uid_t *uid_ptr, gid_t *gid_ptr)
+{
+	const char *colon = strchr(username, ':');
+	if (!colon)
+		return false;
+	long long n, m;
+	n = string_to_uint_ex(username, NULL, 60000, ":");
+	if (n < 0)
+		return false;
+	m = string_to_uint_ex(colon+1, NULL, 60000, NULL);
```

Comment:
I was wondering if there's an appropriate limit for "sane" UID/GID values and after some pondering decided to limit this to what `/etc/login.defs` specifies:

```
# Min/max values for automatic uid selection in useradd
#
UID_MAX			60000
GID_MAX			60000
```

The intent was to not use anything in the sub-uid/gid range by accident.

With the assumption that it is easier to be loosen a limit rather than to tighten it, I wanted to be conservative here.

This can be dropped if you think that strace should not be policing this value.

Reply:
The values configured via login.defs are not set in stone, so it's definitely not a good idea to hardcode these values in strace.  Anyway, given that setpriv(1) from util-linux allows arbitrary uid/gid values, it's not for strace to enforce the policy.

Reply:
Agreed, I'll drop this restriction.

Reply:
Updated